packages/frontend/src/views/CalendarView/CalendarView.tsx

```diff
@@ -0,0 +1,268 @@
+import { useQuery } from '@apollo/client';
+import React, { useState } from 'react';
+import Timematcher from '../../components/Timematcher';
+import SidebarLayout from '../../layouts/SidebarLayout';
+import TitleLayout from '../../layouts/TitleLayout';
+import styles from './CalendarView.module.css';
+import {
+  GET_USER_FLOCK,
+  GetCurrentFlockResult,
+  GET_FLOCK_PARTICIPANTS,
+  GET_USER_CALENDARS,
+  GetCurrentUserResult,
+  GET_USER_INTERVALS,
+  GetUserIntervalsResult,
+  GetFlockIntervalsResult,
+  GET_FLOCK_INTERVALS,
+} from '../../apollo';
+import { CircularProgress } from '@mui/material';
+import { useParams } from 'react-router-dom';
+import ParticipantList from '../../components/ParticipantList';
+import Line from '../../components/Line';
+import { UserAvailabilityPartialDTO, UserIntervalInputDTO } from '@flocker/api-types';
+import CalendarList from '../../components/CalendarList';
+
+type Availability = {
+  start: Date;
+  end: Date;
+  available: boolean;
+};
+
+type Calendar = {
+  name: string;
+  id: string;
+  enabled: boolean;
+  onEnabledChanged: (enabled: boolean) => void;
+};
+
+type Participant = {
+  id: string;
+  name: string;
+};
+
+type FlockParams = {
+  flockCode: string;
+};
+
+const FIFTEEN_MINUTES = 15 * 60000;
+
+type FlockProps = {
+  flockName: string;
+  datesPicked: Date[];
+  timeRange: [Date, Date];
+  userAvailability: Availability[];
+  othersAvailability: Availability[];
+};
+
+const Flock: React.FC<FlockProps> = ({ flockName, datesPicked, timeRange, userAvailability, othersAvailability }) => {
+  return (
+    <TitleLayout
+      title={flockName}
+      content={
+        <>
+          <div className={styles.timeMatcher}>
+            <Timematcher
+              datesPicked={datesPicked}
+              timeRange={timeRange}
+              userAvailability={userAvailability}
+              othersAvailability={othersAvailability}
+            />
+            <Legend />
+          </div>
+        </>
+      }
+    />
+  );
+};
+
+const Legend: React.FC = () => {
+  return (
+    <div>
+      <h2>Legend</h2>
+      <div className={styles.legendKeys}>
+        <div className={styles.circleBothAvailable} />
+        <h3>All available</h3>
+        <div className={styles.circleUserAvailable} />
+        <h3>You're available</h3>
+        <div className={styles.circleFlockAvailable} />
+        <h3>Others available</h3>
+      </div>
+    </div>
+  );
+};
+
+const CalendarView: React.FC = () => {
+  const { flockCode } = useParams<FlockParams>();
+
+  const flock = useQuery<GetCurrentFlockResult>(GET_USER_FLOCK, {
```

Comment:
Related to my comment previous comment. I think there should be an input type inserted after the result type :)

Reply:
fixed :))